docs/en_us/platform_api/source/courses/courses.rst

```diff
@@ -0,0 +1,174 @@
+.. _Courses API Courses Resource:
+
+########################################
+Courses API Courses Resource
```

Comment:
@srpearce and @bderusha Since my last comment, I've talked further with @pdesjardins and I think we have a better idea now, about how to clearly structure our doc for these 2 APIs.
Any of the "catalogs" endpoints *should* live in the catalogs file, contrary to what I was proposing above. 
There is no overlap of the "catalogs" endpoints with the "courses" endpoints, which I'm documenting in files under the "courses" subfolder. The 4 tasks/endpoints I have for the Courses API don't have the concept of catalog at all. 
In the Courses resource:
- Get the details for a specified course (``/api/courses/v1/courses/{course_key}/``)
- Get the list of courses visible to a specified user (``/api/courses/v1/courses/`` with username parm)
In the Blocks resource:
- Get a list of course blocks in a specified course (``/api/courses/v1/blocks/?course_id=<course_id>``)
- Get a list of course blocks in a specific block tree (``/api/courses/v1/blocks/<usage_id>/``)

I'm still a noob at our API docs, so I hope that makes sense? @srpearce when you're back in the office on Tuesday we can meet to review.